scipy/linalg/tests/test_basic.py

```diff
@@ -174,6 +174,17 @@ def test_native_list_arguments(self):
         x = solve_banded((l, u), ab, b)
         assert_array_almost_equal(dot(a, x), b)
 
+    def test_empty(self):
+        # ab contains one empty row corresponding to the diagonal
+        ab = [[]]
+        b = []
+        x = solve_banded((0, 0), ab, b)
+        assert_allclose(x, [])
```

Comment:
```
In [1]: import numpy as np

In [2]: a = np.empty((0, 0))

In [3]: from numpy.testing import assert_allclose

In [4]: assert_allclose(a, 8)     # does not fail!
```

What does the assert_allclose actually check here.
ISTM the check should be `x.shape == (0,)`, what am I missing?

Reply:
I don't think you're missing anything. I guess it didn't occur to me that any of these functions might return a scalar and that this would bypass the shape check in `assert_allclose`. A direct shape check would make more sense. It's been a while since I last looked at this, but I can amend this and similar assertions.

Reply:
Yeah, I fully realize it must be rather frustrating to be asked to do a major change this late! 
Would be very welcome though. My intention is to review/land this PR "soon", and this comment is me starting to work out what is the endgame.

Reply:
Actually, it's not just shape, it's also dtype:

```
In [5]: a32 = np.eye(0, dtype=np.float32)

In [6]: np.linalg.cholesky(a32)
Out[6]: array([], shape=(0, 0), dtype=float32)
```

I _suspect_ the majority of the code itself it fine because of `np.empty_like`, but this definitely needs testing.